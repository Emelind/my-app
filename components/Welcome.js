import React from 'react';
import { Text, Button, View } from 'react-native';
import ColorPicker from './ColorPicker';

const Welcome = ({setShowWelcome}) => {

    return (
        <View>
            <Text 
                style={{
                    padding: 20,
                    fontSize: 20}}
            >
                Welcome to the dog app!
            </Text>
            <Button 
                onPress={() => setShowWelcome(false)}
                title='Check out the dogs!'
                testID='welcomeButton'
            />

        </View>

    );
}

export default Welcome;