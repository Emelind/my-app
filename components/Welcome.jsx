import React from 'react';
import { Text, Button, View } from 'react-native';
import ColorPicker from './ColorPicker';

const Welcome = ({setShowWelcome}) => {

    return (
        <View>
            <Text 
                style={{
                    padding: 20,
                    fontSize: 20}}
            >
                Welcome to the dog app!
            </Text>
            <Button 
                onPress={() => setShowWelcome(false)}
                title='Check out the dogs!'
            />
            <Text
                style={{
                    paddingTop:20}}>
                Pick a color:
            </Text>
            <View 
                style={{
                    flexDirection: 'row',
                    justifyContent: 'space-around',
                    paddingTop: 10}}>
                <ColorPicker color='green'/>
                <ColorPicker color='purple'/>
                <ColorPicker color='pink'/>
                <ColorPicker color='white'/>
            </View>
        </View>

    );
}

export default Welcome;