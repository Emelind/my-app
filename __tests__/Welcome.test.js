import 'react-native';
import React from 'react';
import Welcome from '../components/Welcome';
import {render} from '@testing-library/react-native';

describe('welcome component', () => {
    it('should have a text saying Welcome to the dog app!', () => {
        const welcomeText = 'Welcome to the dog app!';
        const notFoundText = 'Not found text';
        const {toJSON, getByText, queryByText} = render(<Welcome />);
        const foundWelcomeTextElement = getByText(welcomeText);
        const notFoundTextElement = queryByText(notFoundText);
        expect(foundWelcomeTextElement.props.children).toEqual(welcomeText);
        expect(notFoundTextElement).toBeNull();
        expect(toJSON()).toMatchSnapshot();
    });
});
