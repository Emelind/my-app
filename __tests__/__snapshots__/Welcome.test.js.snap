// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`welcome component should have a text saying Welcome to the dog app! 1`] = `
<View>
  <Text
    style={
      Object {
        "fontSize": 20,
        "padding": 20,
      }
    }
  >
    Welcome to the dog app!
  </Text>
  <View
    accessibilityRole="button"
    accessibilityState={Object {}}
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="welcomeButton"
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Check out the dogs!
      </Text>
    </View>
  </View>
</View>
`;
